test/functional/sharding_connection.test.js

```diff
@@ -1,13 +1,30 @@
 'use strict';
 
+const withClient = require('./shared').withClient;
 const setupDatabase = require('./shared').setupDatabase;
 const expect = require('chai').expect;
+const TopologyType = require('../../lib/core/sdam/common').TopologyType;
 
 describe('Sharding (Connection)', function() {
   before(function() {
     return setupDatabase(this.configuration);
   });
 
+  it('Should use sharded topology', {
+    metadata: { requires: { topology: 'sharded' } },
+    test: function() {
+      const client = this.configuration.newClient({}, { useUnifiedTopology: true });
+      expect(client.s.options).to.have.property('useUnifiedTopology', true);
```

Comment:
prefer not to observe private properties, I think it's satisfactory to have the check for the topology description `type` below.